test/e2e/performanceprofile/functests/13_llc/llc.go

```diff
@@ -191,6 +213,347 @@ var _ = Describe("[rfe_id:77446] LLC-aware cpu pinning", Label(string(label.Open
 			})
 		})
 	})
+	Context("Functional Tests", func() {
+		var L3CacheGroupSize int
+		BeforeAll(func() {
+			var (
+				reserved, isolated cpuset.CPUSet
+				policy             = "single-numa-node"
+				onlineCPUSet       cpuset.CPUSet
+			)
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			ctx := context.Background()
+			for _, cnfnode := range workerRTNodes {
+				numaInfo, err := nodes.GetNumaNodes(ctx, &cnfnode)
+				Expect(err).ToNot(HaveOccurred(), "Unable to get numa information from the node")
+				if len(numaInfo) < 2 {
+					Skip(fmt.Sprintf("This test need 2 Numa nodes. The number of numa nodes on node %s < 2", cnfnode.Name))
+				}
+
+				getCCX := nodes.GetL3SharedCPUs(&cnfnode)
+				reserved, err = getCCX(0)
+				Expect(err).ToNot(HaveOccurred())
+				onlineCPUSet, err = nodes.GetOnlineCPUsSet(context.TODO(), &cnfnode)
+				Expect(err).ToNot(HaveOccurred())
+				L3CacheGroupSize = reserved.Size()
+				if len(numaInfo[0]) == L3CacheGroupSize {
+					Skip("This test requires systems where L3 cache is shared amount subset of cpus")
+				}
+			}
+			// Modify the profile such that we give 1 whole ccx to reserved cpus
+			By("Modifying the profile")
+			isolated = onlineCPUSet.Difference(reserved)
+			reservedSet := performancev2.CPUSet(reserved.String())
+			isolatedSet := performancev2.CPUSet(isolated.String())
+			testlog.Infof("Modifying profile with reserved cpus %s and isolated cpus %s", reserved.String(), isolated.String())
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reservedSet,
+				Isolated: &isolatedSet,
+			}
+			profile.Spec.NUMA = &performancev2.NUMA{
+				TopologyPolicy: &policy,
+			}
+			By("Updating Performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By(fmt.Sprintf("Applying changes in performance profile and waiting until %s will start updating", poolName))
+			profilesupdate.WaitForTuningUpdating(ctx, profile)
```

Comment:
this wait is proven fragile though. A better approach seems to be to record the object ResourceVersion, then send the update, then wait for the object to become `Updated` with `resourceVersion != oldResourceVersion`